modules/FiniteSetsExt.tla

```diff
@@ -2,60 +2,59 @@
-    A tool is free to define its own, more efficient, implementation.
-   *) 
-(*  ReduceSet(op, set, base) *)
-  (* sm: alternative, equivalent definition *)
-  FoldMap(op, base, LAMBDA x : x, set)
-
-(***************************************************************************)
-(* Fold over a function (or sequence).                                     *)
-(***************************************************************************)
-FoldFunction(op(_,_), base, fun) == 
-  FoldMap(op, base, LAMBDA i : fun[i], DOMAIN fun)
-
-
-
-(***************************************************************************)
-(* Fold over a subset of the domain of a function -- corresponds to Reduce *)
-(***************************************************************************)
-FoldFunctionOnSet(op(_,_), base, fun, set) ==
-  LET Squeeze(f, S) == [ s \in S |-> f[s]  ]
-  IN
-  FoldFunction(op, base, Squeeze(fun, set))
-
+   (*************************************************************************)
+   (* Fold op over the elements of set using base as starting value.        *)
+   (*                                                                       *)
+   (* Intuitive semantics:                                                  *)
+   (* Initialize a temporary variable `accum` with the value of `base`.     *)
+   (* Iterate over the elements in set in a deterministic but unknown       *)
+   (* order pdate the value of `accum` to the value of `op(accum, x`),      *)
+   (* where `x` s the current value of the iterator.                        *)
+   (*                                                                       *)
+   (* Example:                                                              *)
+   (*         FoldSet(LAMBA x,y : x + y, 0 .. 10, 0) = 55                   *)
+   (*************************************************************************)
+   MapThenFoldSet(op, base, LAMBDA x : x, set)
 
-ReduceSet(op(_, _), set, acc) == FoldSet(op, set, acc)
+Sum(set) ==
+   (*************************************************************************)
+   (* Calculuate the sum of the elements in set.                            *)
+   (*                                                                       *)
+   (* Example:                                                              *)
+   (*         Sum(0 .. 10) = 55                                             *)
+   (*************************************************************************)
+   FoldSet(LAMBDA x, y: x + y, set, 0)
 
-Sum(set) == ReduceSet(LAMBDA x, y: x + y, set, 0)
+Product(set) ==
+   (*************************************************************************)
+   (* Calculuate the product of the elements in set.                        *)
+   (*                                                                       *)
+   (* Example:                                                              *)
+   (*         Sum(0 .. 10) = 55                                             *)
```

Comment:
Copy&Paste error because this is the example for `Sum`.

Reply:
Thanks! Fixed it.